web/src/lib/components/shared-components/autogrow-textarea.svelte

```diff
@@ -0,0 +1,40 @@
+<script lang="ts">
+  import { autoGrowHeight } from '$lib/actions/autogrow';
+  import { clickOutside } from '$lib/actions/click-outside';
+  import { shortcut } from '$lib/actions/shortcut';
+  import { tick } from 'svelte';
+
+  export let content: string = '';
+  export let elementClass: string = '';
+  export let onContentUpdate: (newContent: string) => void = () => null;
+  export let placeholder: string = '';
+
+  let textarea: HTMLTextAreaElement;
+  $: newContent = content;
+
+  $: if (textarea) {
+    newContent;
+    void tick().then(() => autoGrowHeight(textarea));
+  }
+
+  const updateContent = () => {
+    if (content === newContent) {
+      return;
+    }
+    onContentUpdate(newContent);
+  };
+</script>
+
+<textarea
+  bind:this={textarea}
+  class={elementClass}
+  on:focusout={updateContent}
+  on:input={(e) => (newContent = e.currentTarget.value)}
+  use:clickOutside={{ onOutclick: updateContent }}
```

Comment:
`use:clickOutside` seems redundant and sometimes causes updateContent to be called twice

Reply:
Good catch. I couldn't get a test to trigger the function being called twice even when firing a click event on document.body, but I see in testing in a browser that it did.